### DISCUSSION TITLE
Extra Devices Program name and location

### DISCUSSION THREAD

**Discussion by renatoa** [artisan-roaster-scope/artisan#737] [Q&A] [answered]
As title say, where is specified the name and path of the extra device of type "Program" ?

**Comment by MAKOMO** [comment, marked as answer]
Menu Config >> Devices, first tab

<img width="642" alt="Screenshot 2021-11-24 at 20 47 54" src="https://user-images.githubusercontent.com/13909011/143304254-9931da23-3b74-436b-95ff-96ab4f2ba50f.png">

There is a Help page under the Help button

**Comment by renatoa** [comment]
Then, return (move the bullet) to the main device, which is Modbus, as explained in the e-mail ?

**Reply by MAKOMO** [reply]
No program should work as extra device even if the bullet is not selected in the first tab. This is different to the Arduino/TC4 which behaves different for several reasons.

**Comment by renatoa** [comment]
So I can't mix data from a ModBus source with data from other source in Artisan?

**Comment by MAKOMO** [comment]
Of course you can. Just not with Arduino/TC4. "program" works as extra device. You can also connect any number of Phidgets of same or different type.

**Comment by renatoa** [comment]
No TC4 was been mentioned in the opening of this thread. 
Is about a primary data source from a ModBus machine, and an additional measurement I want to add in the mix to be considered by Artisan.
If can't be done by Artisan using an extra device, will write a proxy app that will get data from both sources, and deliver to Artisan as a virtual ModBus with an additional data register.

**Reply by MAKOMO** [reply]
As said, it can be done. Just use the "program" device type and write a small script handing over the data or read the sensor using an appropriate Phidget module (or any other device type supported by Artisan, just not the TC4).

**Comment by bigl00z3** [comment]
What do you mean by “just two digits”? No decimals? What do you have in Config/Curves menu go to the UI tab?
Is the Decimal box ticked?

	

Nico

**Comment by renatoa** [comment]
Deleted my post, because solved, even if I don't know how and why...
During first attempts my program output was like "123,456", without quotes, and the two Extra LCDs shown just two dashes, like --
When output was limited to two digits figures, like 12,78, it worked fine.
Then played a bit with various formats, and no success... until adding a Program 34 device, and suddenly it started working, it shows now 123 and 200, as is ok, for pressure and temperature. Even if Program 34 was been removed meanwhile ! As I wrote, no idea why behaved like that the first time...

No need for decimals so far.

**Reply by MAKOMO** [reply]
Hm, confusing. Please open an issue if you thing you hit an implementation bug with a description that allow us to replicate (and in the best case resolve) the problem.
